### ISSUE TITLE
Summary of How Multipa Works?

### ISSUE THREAD

**Issue by lancejpollard** [ctaguchi/multipa#1]
Hi, I reached out to you via email just now FYI, just introducing myself basically, but wanted to follow up here.

### What is required to support all IPA characteristics across all languages?

I am interested in languages with interesting phonemic patterns. Basically I just would like to cover all the ground of the pronunciation possibilities, in as short a word list as possible.

- Tones (Chinese, Burmese, Yoruba, etc.). Should include single tone words (Yoruba has a lot), and shifting tones (Chinese has 2-tone vowels, Burmese, Thai, and Tibetan have 2-tone and also 3-tone vowels if I remember correctly).
- Clicks (Xhosa is a good candidate, I do a lot with this language). Should cover clicks with and without aspiration, and include all possible clicks. One click is only found in !Xoo, so need at least one word from that.
- Words with palatalization (Russian has a lot of words with this).
- Glottalization (Arabic has this)
- Velarization (Irish has this)
- Pharyngealization (Arabic has this, know of any others?)
- Labialization (Not sure..., some Native American languages I think)
- Aspiration (Hindi or Sanskrit words)
- All vowels (English has many as a good start, but then check out Danish, Swedish, and other European languages with lots of vowels)
- All consonants (English is a good start, Hindi/Sanskrit for retroflex consonants, Navajo for that ɬ sound, Zulu I think for ɮ, french for "back r" ʁ, Hebrew for harsh χ).

**How far is multipa from supporting all these features?** What would it take (what work is there to be done to support these features)? Perhaps I could help in some way.

### Does this work from Text to IPA, or Audio to IPA?

I can't tell but looking briefly at the code it appears this is a text transcription 2 IPA library, not an audio to IPA, is that correcft? Do you have an audio to IPA library elsewhere if correct? If not, what am I missing? (I am looking at things like [finnish_to_ipa.py](https://github.com/ctaguchi/multipa/blob/main/converter/finnish_to_ipa.py) for example).

Basically, how do I do audio to IPA, given your paper 😁 ❤️ 🚀

**Comment by ctaguchi** [comment]
Thank you for your interest. 
## IPA Coverage
For the coverage of IPA symbols, you simply need more training data that have symbols.
Since the model presented in the paper only uses 7 languages, it should be far away from covering all the possible phonetic sounds; rather, our model is design to give us an approximated IPA, so that it can produce a draft transcription that mitigates linguists/phoneticians' workload.

## model
For languages that lack datasets with phonetic IPA transcription, we approximate it by creating a pseudo-phonetic dataset with rule-based grapheme-to-phoneme conversion. That's what those text-to-IPA scripts are doing. 
As you can see in the repository, the main program provides example code to train an audio-to-IPA model. The link to a trained model is given in README.md.
